includes/controls/settings-footer.php

```diff
@@ -0,0 +1,535 @@
+		);
+
+		$this->add_control(
+			'hello_footer_copyright_color',
+			[
+				'label' => __( 'Text Color', 'hello-elementor' ),
+				'type' => Controls_Manager::COLOR,
+				'condition' => [
+					'hello_footer_copyright_display' => 'yes',
+				],
+				'selectors' => [
+					'.site-footer .copyright p' => 'color: {{VALUE}};',
+				],
+			]
+		);
+
+		$this->add_group_control(
+			\Elementor\Group_Control_Typography::get_type(),
+			[
+				'name' => 'hello_footer_copyright_typography',
+				'label' => __( 'Typography', 'hello-elementor' ),
+				'condition' => [
+					'hello_footer_copyright_display' => 'yes',
+				],
+				'selector' => '.site-footer .copyright p',
+			]
+		);
+
+		$this->end_controls_section();
+	}
+
+	public function on_save( $data ) {
+		// Save chosen footer menu to the WP settings.
+		if ( isset( $data['settings']['hello_footer_menu'] ) ) {
+			$menu_id = $data['settings']['hello_footer_menu'];
+			$locations = get_theme_mod( 'nav_menu_locations' );
+			$locations['menu-2'] = (int) $menu_id;
+			set_theme_mod( 'nav_menu_locations', $locations );
+		}
+	}
+
+	public function get_additional_tab_content() {
+		if ( ! defined( 'ELEMENTOR_PRO_VERSION' ) ) {
+			return wp_sprintf( '
+				<div class="elementor-nerd-box">
+					<img src="%4$s" class="elementor-nerd-box-icon">
+					<div class="elementor-nerd-box-message">
+						<p class="elementor-panel-heading-title elementor-nerd-box-title">%1$s</p>
+						<p>%2$s</p>
+					</div>
+					<a class="elementor-button elementor-button-default elementor-nerd-box-link" target="_blank" href="https://elementor.com/pro/?utm_source=panel-widgets&amp;utm_campaign=gopro&amp;utm_medium=wp-dash&amp;utm_term=helloelementor">%3$s</a>
+				</div>
+				',
+				__( 'Create a custom footer with multiple options', 'hello-elementor' ),
+				__( 'Upgrade to Elementor Pro and enjoy free design and many more features', 'hello-elementor' ),
+				__( 'Go Pro', 'hello-elementor' ),
+				get_template_directory_uri() . '/assets/images/go-pro.svg'
+			);
+		} else {
+			return wp_sprintf( '
+				<div class="elementor-nerd-box">
+					<img src="%4$s" class="elementor-nerd-box-icon">
+					<div class="elementor-nerd-box-message">
+						<p class="elementor-panel-heading-title elementor-nerd-box-title">%1$s</p>
+						<p class="elementor-nerd-box-message">%2$s</p>
+					</div>
+					<a class="elementor-button elementor-button-success elementor-nerd-box-link" target="_blank" href="%5$s">%3$s</a>
+				</div>
+				',
+				__( 'Create a custom header with the new Theme Builder', 'hello-elementor' ),
+				__( 'With the new Theme Builder you can jump directly into each part of your site', 'hello-elementor' ),
+				__( 'Create Footer', 'hello-elementor' ),
+				get_template_directory_uri() . '/assets/images/go-pro.svg',
+				get_admin_url( null, 'admin.php?page=elementor-app#/site-editor/templates/footer' )
+			);
+		}
+	}
+}
+
+add_action( 'elementor/kit/register_tabs', function( \Elementor\Core\Kits\Documents\Kit $kit ) {
```

Comment:
?

Reply:
Files containing a class declaration should include **only** the class declaration. Any logic using that class should be included outside this file, for example after the include statement.

Reply:
Moved out of the class file